client/task-list/tasks/payments/index.js

```diff
@@ -42,36 +42,34 @@ class Payments extends Component {
 			( method ) => ( enabledMethods[ method.key ] = method.isEnabled )
 		);
 		this.state = {
+			busyMethod: null,
 			enabledMethods,
+			recommendedMethod: this.getRecommendedMethod(),
 		};
 
 		this.completeTask = this.completeTask.bind( this );
 		this.markConfigured = this.markConfigured.bind( this );
 		this.skipTask = this.skipTask.bind( this );
 	}
 
-	componentDidUpdate( prevProps ) {
-		if ( prevProps === this.props ) {
-			return;
-		}
-		const { methods } = this.props;
-
-		let recommendedMethod = 'stripe';
-		methods.forEach( ( method ) => {
-			const { key, visible } = method;
-
-			if ( key === 'wcpay' && visible ) {
-				recommendedMethod = 'wcpay';
-			}
-		} );
+	componentDidUpdate() {
+		const { recommendedMethod } = this.state;
 
-		if ( this.state.recommendedMethod !== recommendedMethod ) {
+		const method = this.getRecommendedMethod();
+		if ( recommendedMethod !== method ) {
 			this.setState( {
-				recommendedMethod,
+				recommendedMethod: method,
 			} );
 		}
 	}
 
+	getRecommendedMethod() {
+		const { methods } = this.props;
+		return (
+			methods.find( ( m ) => m.key === 'wcpay' && m.visible ) || 'stripe'
```

Comment:
This will return either `'stripe'` or the WCPay method **object**. Fix:
```suggestion
			methods.find( ( m ) => m.key === 'wcpay' && m.visible ) ? 'wcpay' : 'stripe';
```

Reply:
Nice!  Thanks, @DanReyLop!